zavod/zavod/helpers/sanctions.py

```diff
@@ -45,16 +56,25 @@ def make_sanction(
         sanction.add("country", dataset.publisher.country)
     sanction.add("authority", dataset.publisher.name)
     sanction.add("sourceUrl", dataset.url)
-    if program is not None:
-        sanction.set("program", program)
+
+    if program_name is not None:
+        sanction.set("program", program_name)
+
     if program_key is not None:
-        program_id = context.lookup_value("sanction.program", program_key)
-        if program_id is not None:
-            program_url = f"https://www.opensanctions.org/programs/{program_id}"
-            sanction.add("programUrl", program_url)
-            sanction.add("programId", program_id)
+        program = programs.get_program_by_key(context, program_key)
+        if program is not None:
+            sanction.add("programId", program_key)
+            if not program_name:
+                sanction.set("program", program.title)
+            if program.url:
+                sanction.add("programUrl", program.url)
+            else:
+                sanction.add(
+                    "programUrl",
+                    f"https://www.opensanctions.org/programs/{program_key}",
+                )
         else:
-            context.log.warn(f"Program key {program_key!r} not found.", program=program)
+            context.log.warn(f"Program key {program_key!r} not found.")
```

Comment:
so

- if program_name is provided, `Sanction:program` is set to that
- if program_key is provided, `Sanction:program` is set to `program.title` unless `program_name` was provided

This feels like the sort of finicky but important logic where a test might save someone from an unintended mistake

Reply:
I simplified the logic. Now we do this:

- If `program_key` is given, check if it's valid. If it is:
    - Stuff it in `Sanction:programId`
    - Stuff the `Program.url` into `Sanction:programUrl`

I also added a test for this.